eng/templates/azure-pipelines-shared-jobs.yml

```diff
@@ -0,0 +1,134 @@
+              artifact: ProductConstructionService.ScenarioTests
+              displayName: Publish E2E Tests
+            - output: pipelineArtifact
+              path: $(Build.SourcesDirectory)\artifacts\bin\ProductConstructionService.Cli\$(_BuildConfig)\net8.0
+              artifact: ProductConstructionService.Cli
+              displayName: Publish PCS CLI
+        - job: BuildAndPublishDocker
+          displayName: Build/publish docker image
+          pool:
+            name: NetCore1ESPool-Internal
+            demands: ImageOverride -equals 1es-ubuntu-2204
+            os: linux
+          steps:
+          - checkout: self
+          - template: /eng/templates/steps/docker-build.yml@self
+            parameters:
+              devBranchSuffix: $(devBranchSuffix)
+              dockerImageName: $(dockerRegistryUrl)/$(containerName)
+          - ${{ if notin(variables['Build.Reason'], 'PullRequest') }}:
+            - task: AzureCLI@2
+              inputs:
+                azureSubscription: $(serviceConnectionName)
+                scriptType: pscore
+                scriptLocation: inlineScript
+                inlineScript: |
+                  az acr login --name $(containerRegistryName)
+                  docker push "$(dockerRegistryUrl)/$(containerName):$(DockerTag.newDockerImageTag)"
+              displayName: Push docker image
+  - template: /eng/templates/stages/deploy.yaml@self
+    parameters:
+      isProd: ${{ eq(variables['Build.SourceBranch'], 'refs/heads/production') }}
+  - ${{ if in(variables['Build.SourceBranch'], 'refs/heads/main', 'refs/heads/production') }}:
+      - ${{ if eq(parameters.isOfficial, true) }}:
+          - template: /eng/common/templates-official/post-build/post-build.yml@self
+            parameters:
+              enableSymbolValidation: true
+              enableSigningValidation: false
+              artifactsPublishingAdditionalParameters: '/p:CheckEolTargetFramework=false'
+              symbolPublishingAdditionalParameters: '/p:CheckEolTargetFramework=false'
+              SDLValidationParameters:
+                enable: true
+                params: |
+                  -SourceToolsList @("policheck","credscan")
+                  -TsaInstanceURL $(_TsaInstanceURL)
+                  -TsaProjectName $(_TsaProjectName)
+                  -TsaNotificationEmail $(_TsaNotificationEmail)
+                  -TsaCodebaseAdmin $(_TsaCodebaseAdmin)
+                  -TsaBugAreaPath $(_TsaBugAreaPath)
+                  -TsaIterationPath $(_TsaIterationPath)
+                  -TsaRepositoryName "Arcade-Services"
+                  -TsaCodebaseName "Arcade-Services"
+                  -TsaPublish $True
+                  -PoliCheckAdditionalRunConfigParams @("UserExclusionPath < $(Build.SourcesDirectory)/eng/PoliCheckExclusions.xml")
+      - ${{ else }}:
+          - template: /eng/common/templates/post-build/post-build.yml@self
```

Comment:
The arguments are the exact same while the only difference is the template path. Can we maybe simplify this a bit?